src/formats/tana/tana-import.ts

```diff
@@ -0,0 +1,306 @@
+			this.importLibraryNode(libraryNode);
+		}
+		else {
+			this.notices.push('Library node not found');
+		}
+
+		for (let suffix of ['_TRASH', '_SCHEMA', '_SIDEBAR_AREAS', '_USERS', '_SEARCHES', '_MOVETO', '_WORKSPACE', '_QUICK_ADD']) {
+			const specialNode = this.nodes.get(rootNode.id + suffix);
+			if (specialNode != null) {
+				this.markSeen(specialNode);
+			}
+			else {
+				this.notices.push('Special node ' + suffix + ' not found');
+			}
+		}
+
+		this.enumerateChildren(workspaceNode, (childNode) => {
+			if (childNode.props._docType == 'journal') {
+				this.importDailyNotes(childNode);
+			}
+		});
+
+		for (const [node, file] of this.topLevelNodes.values()) {
+			this.convertNode(node, file);
+		}
+
+		this.notices.push('Converted ' + this.convertedNodes.size + ' nodes');
+		let unconverted = 0;
+		for (let node of this.tanaDatabase.docs) {
+			if (!this.convertedNodes.has(node.id) && !node.id.startsWith('SYS') &&
+				node.props._docType != 'workspace') {
+				const path = this.pathFromRoot(node, rootNode);
+				if (path) {
+					this.notices.push('Found unconverted node: ' + path);
+					unconverted++;
+					if (unconverted == 50) break;
+				}
+			}
+		}
+	}
+
+	private prepareAnchors(node: TanaDoc) {
+		if (node.props.name) {
+			for (let m of node.props.name.matchAll(inlineRefRegex)) {
+				this.anchors.add(m[1]);
+			}
+		}
+		this.enumerateChildren(node, (childNode) => {
+			if (childNode.props._ownerId != node.id) {
+				this.anchors.add(childNode.id);
+			}
+			this.prepareAnchors(childNode);
+		});
+	}
+
+	private importDailyNotes(node: TanaDoc) {
+		this.convertedNodes.add(node.id);
+		this.enumerateChildren(node, (yearNode) => {
+			this.convertedNodes.add(yearNode.id);
+			this.enumerateChildren(yearNode, (weekNode) => {
+				this.convertedNodes.add(weekNode.id);
+				this.enumerateChildren(weekNode, (dayNode) => {
+					if (dayNode.props.name) {
+						this.topLevelNodes.set(dayNode.id, [dayNode, dayNode.props.name]);
+					}
+				});
+			});
+		});
+	}
+
+	private importLibraryNode(node: TanaDoc) {
+		this.convertedNodes.add(node.id);
+		this.enumerateChildren(node, (childNode) => {
+			this.topLevelNodes.set(childNode.id, [childNode, childNode.props.name]);
+		});
+	}
+
+	private convertNode(node: TanaDoc, filename: string) {
+		const fragments: Array<string> = [];
+		const properties = this.collectNodeProperties(node);
+		if (properties.length > 0) {
+			fragments.push('---');
+			for (const [, k, v] of properties) {
+				fragments.push(k + ': ' + v);
+			}
+			fragments.push('---');
+		}
+		this.convertNodeRecursive(node, fragments, 0);
+		this.result.set(filename + '.md', fragments.join('\n'));
+	}
+
+	private collectNodeProperties(node: TanaDoc): Array<[string, string, string]> {
+		const properties: Array<[string, string, string]> = [];
+		this.enumerateChildren(node, (child) => {
+			if (child.props._docType == 'tuple' && child.children.length >= 2) {
+				const propNode = this.nodes.get(child.children[0]);
```

Comment:
Am I reading correctly that properties for a node are stored as separate `TanaDoc` instances?